commons/all/src/main/java/org/infinispan/commons/dataconversion/internal/Json.java

```diff
@@ -1189,6 +1190,8 @@ else if (anything instanceof Collection<?>) {
             return O;
          } else if (anything instanceof Boolean)
             return factory().bool((Boolean) anything);
+         else if (anything instanceof TimeQuantity)
+            return factory().string(anything.toString());
```

Comment:
I guess this is fine, as we expose the value as is. But this would create a little more effort on the consuming side to implement the parsing, too.